clients/src/main/java/org/apache/kafka/common/record/EndTransactionMarker.java

```diff
@@ -95,32 +71,28 @@ public int hashCode() {
 
     private static void ensureTransactionMarkerControlType(ControlRecordType type) {
         if (type != ControlRecordType.COMMIT && type != ControlRecordType.ABORT)
-            throw new IllegalArgumentException("Invalid control record type for end transaction marker" + type);
+            throw new IllegalArgumentException("Invalid control record type for end transaction marker " + type);
     }
 
     public static EndTransactionMarker deserialize(Record record) {
         ControlRecordType type = ControlRecordType.parse(record.key());
         return deserializeValue(type, record.value());
     }
 
+    // Visible for testing
     static EndTransactionMarker deserializeValue(ControlRecordType type, ByteBuffer value) {
         ensureTransactionMarkerControlType(type);
 
-        if (value.remaining() < CURRENT_END_TXN_MARKER_VALUE_SIZE)
-            throw new InvalidRecordException("Invalid value size found for end transaction marker. Must have " +
-                    "at least " + CURRENT_END_TXN_MARKER_VALUE_SIZE + " bytes, but found only " + value.remaining());
-
-        short version = value.getShort(0);
-        if (version < 0)
```

Comment:
Should we keep the version check to protect us from reading unknown version in the future?

Reply:
Thank you, I thought it was trivial and deleted this, in fact this check makes sense as you suggested, I have changed it back.